x-pack/test/api_integration/apis/ml/data_frame_analytics/delete.ts

```diff
@@ -0,0 +1,237 @@
+import { DataFrameAnalyticsConfig } from '../../../../../plugins/ml/public/application/data_frame_analytics/common';
+import { DeepPartial } from '../../../../../plugins/ml/common/types/common';
+const COMMON_HEADERS = {
+  'kbn-xsrf': 'some-xsrf-token',
+};
+
+export default ({ getService }: FtrProviderContext) => {
+  const es = getService('legacyEs');
+  const esArchiver = getService('esArchiver');
+  const supertest = getService('supertestWithoutAuth');
+  const ml = getService('ml');
+
+  const jobId = `bm_${Date.now()}`;
+  const generateDestinationIndex = (analyticsId: string) => `user-${analyticsId}_1`;
+  const commonJobConfig = {
+    source: {
+      index: ['ft_bank_marketing'],
+      query: {
+        match_all: {},
+      },
+    },
+    analysis: {
+      classification: {
+        dependent_variable: 'y',
+        training_percent: 20,
+      },
+    },
+    analyzed_fields: {
+      includes: [],
+      excludes: [],
+    },
+    model_memory_limit: '350mb',
+  };
+
+  const testJobConfigs: Array<DeepPartial<DataFrameAnalyticsConfig>> = [
+    'Test delete job only',
+    'Test delete job and target index',
+    'Test delete job and index pattern',
+    'Test delete job, target index, and index pattern',
+  ].map((description, idx) => {
+    const analyticsId = `${jobId}_${idx + 1}`;
+    return {
+      id: analyticsId,
+      description,
+      dest: {
+        index: generateDestinationIndex(analyticsId),
+        results_field: 'ml',
+      },
+      ...commonJobConfig,
+    };
+  });
+
+  async function createJobs(mockJobConfigs: Array<DeepPartial<DataFrameAnalyticsConfig>>) {
+    for (const jobConfig of mockJobConfigs) {
+      await ml.api.createDataFrameAnalyticsJob(jobConfig as DataFrameAnalyticsConfig);
+    }
+  }
+
+  async function jobDeleted(analyticsId: string) {
+    return await supertest
+      .get(`/api/ml/data_frame/analytics/${analyticsId}`)
+      .auth(USER.ML_POWERUSER, ml.securityCommon.getPasswordForUser(USER.ML_POWERUSER))
+      .set(COMMON_HEADERS)
+      .expect(404);
+  }
+  describe('DELETE data_frame/analytics', () => {
+    before(async () => {
+      await esArchiver.loadIfNeeded('ml/bm_classification');
+      await ml.testResources.setKibanaTimeZoneToUTC();
+      await createJobs(testJobConfigs);
+    });
+
+    after(async () => {
+      await ml.api.cleanMlIndices();
+    });
+
+    describe('DeleteDataFrameAnalytics', () => {
+      it('should delete analytics jobs by id', async () => {
+        const { body } = await supertest
+          .delete(`/api/ml/data_frame/analytics/${jobId}_1`)
+          .auth(USER.ML_POWERUSER, ml.securityCommon.getPasswordForUser(USER.ML_POWERUSER))
+          .set(COMMON_HEADERS)
+          .expect(200);
+        expect(body.analyticsJobDeleted.success).to.eql(true);
+      });
+
+      it('should not allow to retrieve analytics jobs for the user without required permissions', async () => {
```

Comment:
I would split this one in two test (one for `ml_unauthorized` and one for `ml_viewer`), so in an error case it's more clear which one failed.